lib/src/main/java/com/thenewboston/api/bankapi/datasource/BankDataSource.kt

```diff
@@ -125,12 +130,35 @@ class BankDataSource @Inject constructor(private val networkClient: NetworkClien
         }
     }
 
-    suspend fun sendBankTrust(request: BankTrustRequest) = makeApiCall(
-        call = { doBankTrust(request) },
+    suspend fun updateAccountTrust(accountNumber: String, request: UpdateTrustRequest): Outcome<Account> = makeApiCall(
+        call = { doUpdateAccount(accountNumber, request) },
+        errorMessage = "Could not update trust level of given account"
+    )
+
+    private suspend fun doUpdateAccount(accountNumber: String, request: UpdateTrustRequest): Outcome<Account> {
+        val patchAccountUrl = "${BankAPIEndpoints.ACCOUNTS_ENDPOINT}/$accountNumber"
+
+        val updatedAccount = networkClient.defaultClient.patch<Account> {
+            url(patchAccountUrl)
+            contentType(ContentType.Application.Json)
+            body = request
+        }
+
+        return when {
+            updatedAccount.id.isBlank() -> Outcome.Error(
+                "Received unexpected response when updating trust level of account $accountNumber",
+                IOException()
+            )
+            else -> Outcome.Success(updatedAccount)
+        }
+    }
+
+    suspend fun updateBankTrust(request: UpdateTrustRequest): Outcome<BankTrustResponse> = makeApiCall(
+        call = { doUpdateBankTrust(request) },
         errorMessage = "Could not send bank trust for ${request.nodeIdentifier}"
     )
 
-    private suspend fun doBankTrust(request: BankTrustRequest): Outcome<BankTrustResponse> {
+    private suspend fun doUpdateBankTrust(request: UpdateTrustRequest): Outcome<BankTrustResponse> {
```

Comment:
We can remove the BankTrustRequest data class since we won't be needing it anymore.

Reply:
Good catch, removed the `BankTrustRequest`.